IOS Practicals New/ViewControllers/UICompSignUpViewController.swift

```diff
@@ -0,0 +1,150 @@
+        }
+    }
+    
+    @IBAction func radioBtnGender(_ sender: UIButton) {
+        if sender.tag == 1 {
+            radioBtnMale.isSelected = true
+            radioBtnFemale.isSelected = false
+            print("Selected Male")
+        } else if sender.tag == 2 {
+            radioBtnMale.isSelected = false
+            radioBtnFemale.isSelected = true
+            print("Selected Female")
+        } else {
+            print("Selected Nothing")
+        }
+    }
+    
+} // End of class
+
+// MARK: - UIImagePickerControllerDelegate
+extension UICompSignUpViewController: UIImagePickerControllerDelegate, UINavigationControllerDelegate {
+    
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey : Any]) {
+        if let image = info[UIImagePickerController.InfoKey(rawValue: "UIImagePickerControllerOriginalImage")] as? UIImage {
+            imgViewProfile.image = image
+        }
+        picker.dismiss(animated: true, completion: nil)
+    }
+    
+    func imagePickerControllerDidCancel(_ picker: UIImagePickerController) {
+        picker.dismiss(animated: true, completion: nil)
+    }
+    
+} // End of Extension
+
+// MARK: - UITextFieldDelegate
+extension UICompSignUpViewController: UITextFieldDelegate{
+    
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        switch textField {
+        case self.tfEmail:
+            self.tfPassword.becomeFirstResponder()
+        case self.tfPassword:
+            self.tvBio.becomeFirstResponder()
+        default:
+            self.resignFirstResponder()
+        }
+        return true
+    }
+    
+} // End of Extension
+
+// MARK: - UITextView
+extension UITextView {
+    
+    func addDoneButtonOnKeyboard() {
+        let keyBoardToolBar = UIToolbar()
+        keyBoardToolBar.sizeToFit()
+        let flexibleSpace = UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+        let doneButton = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(resignFirstResponder))
+        keyBoardToolBar.items = [flexibleSpace, doneButton]
+        self.inputAccessoryView = keyBoardToolBar
+    }
+    
+} // End of Extension
+
+// MARK: - File Private Functions
+extension UICompSignUpViewController {
+    
+    private func initialData() {
+        let borderColor : UIColor = UIColor(red: 0.85, green: 0.85, blue: 0.85, alpha: 1.0)
+        tvBio.layer.borderWidth = 0.5
+        tvBio.layer.borderColor = borderColor.cgColor
+        tvBio.layer.cornerRadius = 5.0
+        tvBio.addDoneButtonOnKeyboard()
+        
+        imgViewProfile.layer.borderWidth = 1
+        imgViewProfile.layer.borderColor = UIColor.black.cgColor
+        imgViewProfile.layer.cornerRadius = imgViewProfile.frame.height/2
+        
+        let tapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(imageTapped(tapGestureRecognizer:)))
```

Comment:
It should be great if you use UIViewController extension to write this tapgesture code so you don't need to write code again and again